extensions/wikia/EditPageLayout/js/plugins/FlowTracking.js

```diff
@@ -0,0 +1,43 @@
+(function(window,$){
+
+	var WE = window.WikiaEditor = window.WikiaEditor || (new Observable());
+
+	var editorName = function(mode) {
+		var RTE = (window.RTE === undefined);
+
+		if (!RTE) {
+			return 'rte-' + mode;
+		} else if (mode == 'source') {
+			return 'sourceedit';
+		}
+
+		return mode;
+	};
+
+	WE.plugins.flowtracking = $.createClass(WE.plugin,{
+
+		initEditor: function(editor) {
+			require(['wikia.flowTracking', 'wikia.querystring', 'mw'], function(flowTrack, QueryString, mw) {
+				var namespaceId = mw.config.get('wgNamespaceNumber'),
+					articleId = mw.config.get('wgArticleId');
+
+				// Track only creating articles (wgArticleId=0) from namespace 0 (Main)
+				// IMPORTANT: on Special:CreatePage even after providing article title the namespace is set to -1 (Special Page)
+				if (namespaceId === 0 && articleId === 0) {
+					var qs = new QueryString(window.location.href);
+
+					// 'flow' is the parameter passed in the url if user has started a flow already
+					var flowParam = qs.getVal('flow', false);
+
+					if (flowParam || document.referrer) {
+						//TODO: track middle step for other flows
+					} else {
+						flowTrack.beginFlow('direct-url', {editor: editorName(editor.mode)});
+						console.log(editorName(editor.mode));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this logging statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=javascript%3AS2228)